src/AppBundle/Command/DeleteUserCommand.php

```diff
@@ -0,0 +1,204 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            // a good practice is to use the 'app:' prefix to group all your custom application commands
+            ->setName('app:delete-user')
+            ->setDescription('Deletes users from the database')
+            ->setHelp($this->getCommandHelp())
+            // commands can optionally define arguments and/or options (mandatory and optional)
+            // see http://symfony.com/doc/current/components/console/console_arguments.html
+            ->addArgument('username', InputArgument::REQUIRED, 'The username of an existing user')
+        ;
+    }
+
+    /**
+     * This method is executed before the interact() and the execute() methods.
+     * It's main purpose is to initialize the variables used in the rest of the
+     * command methods.
+     *
+     * Beware that the input options and arguments are validated after executing
+     * the interact() method, so you can't blindly trust their values in this method.
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->entityManager = $this->getContainer()->get('doctrine')->getManager();
+    }
+
+    /**
+     * This method is executed after initialize() and before execute(). Its purpose
+     * is to check if some of the options/arguments are missing and interactively
+     * ask the user for those values.
+     *
+     * This method is completely optional. If you are developing an internal console
+     * command, you probably should not implement this method because it requires
+     * quite a lot of work. However, if the command is meant to be used by external
+     * users, this method is a nice way to fall back and prevent errors.
+     */
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        if (null !== $input->getArgument('username')) {
+            return;
+        }
+
+        // multi-line messages can be displayed this way...
+        $output->writeln('');
+        $output->writeln('Delete User Command Interactive Wizard');
+        $output->writeln('-----------------------------------');
+
+        // ...but you can also pass an array of strings to the writeln() method
+        $output->writeln(array(
+            '',
+            'If you prefer to not use this interactive wizard, provide the',
+            'arguments required by this command as follows:',
+            '',
+            ' $ php app/console app:delete-user username',
+            '',
+        ));
+
+        $output->writeln(array(
+            '',
+            'Now we\'ll ask you for the value of all the missing command arguments.',
+            '',
+        ));
+
+        // See http://symfony.com/doc/current/components/console/helpers/questionhelper.html
+        $console = $this->getHelper('question');
+
+        $username = $input->getArgument('username');
+        if (null === $username) {
```

Comment:
This check is needless. We would already have returned earlier before if the `username` argument were passed.